### PR TITLE
test(): text dnd

### PR DESCRIPTION
**Pull request by ShaMan123**
## Motivation

Testing the formerly untestable drag and drop feature.
After a lot of hours I finally managed to get it to work.
Eraser is next
Playwright is powerful - it was a very good decision

## Changes

- removed `e2e/imports.ts` in favor of a full importmap produced in runtime
- made babelrc esm
- added tests

### CODE REVIEW

**Comment by ShaMan123** [comment]
There is an issue with the fonts

**Comment by asturur** [comment]
> There is an issue with the fonts

We need to be aware of choosing fonts that are available under linux all the time.

Said so, i would like the screenshot to be named clearly.

Like drop-fabric-between-lor-and-emipsum

something like that.
I find very confusing the a-b thing to follow.
Eventually with a number in front.
1)drop-fabic-between
2)drag-image-of...

**Comment by ShaMan123** [comment]
relevant https://wiki.ubuntu.com/Fonts

File: e2e/tests/text/drag&drop/index.spec.ts-snapshots/drag-image-fabric.png
Comment (ShaMan123) [inline e2e/tests/text/drag&drop/index.spec.ts-snapshots/drag-image-fabric.png:1]: this snapshot is failing - help - I tried to remove retina scaling but it didn't help - maybe I didn't do it correctly


e8a69e28dfedfd5a8a3af9a84ec4a8d25e9ec2bc

d692f95a459c844d31b1868aad409dfa4175058a


65e534bd1b803084521e532384b3464e957e5f70

**Review by ShaMan123 — COMMENTED** [review]
Almost done - I need some help to wrap this up

**Review by ShaMan123 — COMMENTED** [review]
- removed `e2e/imports.ts` in favor of a full importmap produced in runtime
- made babelrc esm

File: e2e/tests/text/drag&drop/index.spec.ts [hunk 1]
```diff
@@ -130,6 +130,9 @@ async function waitForDragImage(
 }
 
 test('Drag Image A', async ({ page }) => {
+  // we need to fix font diff
+  process.env.CI && test.fail();
```
Comment (ShaMan123) [inline e2e/tests/text/drag&drop/index.spec.ts:134]: disabled the error for now

File: e2e/setup/setupCoverage.ts [hunk 2]
```diff
@@ -7,28 +7,37 @@ import v8toIstanbul from 'v8-to-istanbul';
...
+          });
+          await converter.load();
+          converter.applyCoverage(functions);
+          return [pathTo, converter.toIstanbul()];
+        })
+      )
+    ).filter(
+      ([pathTo]: [string]) =>
+        !coverageIgnore.some((ignore) =>
+          pathTo.startsWith(path.resolve(process.cwd(), ignore))
```
Comment (ShaMan123) [inline e2e/setup/setupCoverage.ts:39]: ignoring unnecessary coverage data

**Comment by ShaMan123** [comment]
I think we can merge this and get back to the font inconsistency

**Review by ShaMan123 — COMMENTED** [review]
Implemented font caching for perf
I can extract the changes to the test suite to another PR if you want

**Review by ShaMan123 — COMMENTED** [review]
READY
Not touching anymore

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
great test

**Comment by ShaMan123** [comment]
I wish you not delete setupFonts and setupImports
Why remove setupImports?
And for setupFonts, can you leave it there but don't import it so we don't need to redo it with a comment or whatever for the future

**Comment by asturur** [comment]
i didn't delete them, are in a backup branch.
`test-text-dnd-previous`

setupImports was added to add fontObserver? The test work without the change, so the change wasn't needed.
there was no explanation of what setupImports did apart `removed e2e/imports.ts in favor of a full importmap produced in runtime` is completely unclear why to do that since all test were running fine without. 
The moment we need a second import outside fabric in a test we can add it.

setupFonts wasn't good imho It used google fonts and used fontFaceObserver, and i don't want to use either of them.
it had network dependencies on tests, fiddling with google css, network intercepting to make custom chaching. we shouldn't deal with those things. As soon as we need a custom font for a test that is not Arial we can add the font to the repo and load it from localhost using the standard browser api

**Pull request merged**